tests/test-integration/src/test/java/org/eclipse/jetty/test/CustomRequestLogTest.java

```diff
@@ -88,24 +102,70 @@ void testHandlerServerStart(String formatString) throws Exception
         TestRequestLogWriter writer = new TestRequestLogWriter();
         _log = new CustomRequestLog(writer, formatString);
         _server.setRequestLog(_log);
-        _server.setHandler(new TestHandler());
+        ServletContextHandler contextHandler = new ServletContextHandler();
+        contextHandler.setSecurityHandler(getSecurityHandler("username", "password", "testRealm"));
+        contextHandler.addServlet(new ServletHolder(new TestServlet()), "/");
+        _server.setHandler(contextHandler);
         _server.start();
 
         String host = _serverConnector.getHost();
         if (host == null)
-        {
             host = "localhost";
-        }
+
         int localPort = _serverConnector.getLocalPort();
         _serverURI = new URI(String.format("http://%s:%d/", host, localPort));
     }
 
+    private static SecurityHandler getSecurityHandler(String username, String password, String realm)
+    {
+        HashLoginService loginService = new HashLoginService();
+        UserStore userStore = new UserStore();
+        userStore.addUser(username, Credential.getCredential(password), new String[]{"user"});
+        loginService.setUserStore(userStore);
+        loginService.setName(realm);
+
+        Constraint constraint = new Constraint();
+        constraint.setName("auth");
+        constraint.setAuthenticate(true);
+        constraint.setRoles(new String[]{"**"});
+
+        ConstraintMapping mapping = new ConstraintMapping();
+        mapping.setPathSpec("/secure/*");
+        mapping.setConstraint(constraint);
+
+        ConstraintSecurityHandler security = new ConstraintSecurityHandler();
+        security.addConstraintMapping(mapping);
+        security.setAuthenticator(new BasicAuthenticator());
+        security.setLoginService(loginService);
+
+        return security;
+    }
+
     @AfterEach
     public void after() throws Exception
     {
         _server.stop();
     }
 
+    @Test
+    public void testLogRemoteUser() throws Exception
+    {
+        String authHeader = HttpHeader.AUTHORIZATION + ": Basic " + Base64.getEncoder().encodeToString("username:password".getBytes());
+        testHandlerServerStart("%u %{d}u");
+
+        _connector.getResponse("GET / HTTP/1.0\n\n\n");
+        String log = _entries.poll(5, TimeUnit.SECONDS);
+        assertThat(log, is("- -"));
+
+        _connector.getResponse("GET / HTTP/1.0\n" + authHeader + "\n\n\n");
+        log = _entries.poll(5, TimeUnit.SECONDS);
+        assertThat(log, is("- ?username"));
```

Comment:
Do you think we should be appending this `?` before the remote user if it was deferred? Seems like a strange thing to do.

Reply:
Probably not.  Why not have another format %u{D} that either does have the '?'